src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -460,7 +460,8 @@ post_subject_versions(server::request_t rq, server::reply_t rp) {
       unparsed.id.value_or(invalid_schema_id),
       is_deleted::no};
 
-    auto ids = co_await rq.service().schema_store().get_schema_version(schema);
+    auto ids = co_await rq.service().schema_store().get_schema_version(
+      schema.share());
```

Comment:
>     auto ids = co_await rq.service().schema_store().get_schema_version(
>      schema.share());

then below

>         schema_id = co_await rq.service().writer().write_subject_version(
>          std::move(schema));

@BenPope looks ok, but it makes me slightly nervous... do either of these hop cores, hold on to references, and/or mutate the iobuf data?

the cross-core issue is related to reference counting. if nothing is happening cross-core then it's ok.

otherwise, sharing an iobuf is fine, but it isn't copy-on-write so it can't be done naively/in-all-cases. it involves some code inspection to ensure that the new thing receiving the share doesn't expect to own the content of the iobuf.